File: Python/Arrays/secondLargestElementInTheArray.py
def secondLargestElementInTheArray(arr):
    temp = -1
    res = -2
    for i in range(len(arr)):
        if(temp < arr[i]):
            res = temp
            temp = arr[i]
        elif(res < arr[i]):
            res = arr[i]
    return res

if __name__ == "__main__":
    n = int(input("Enter the size of the array : "))
    arr = list()
    print("Enter the elements of the array : ")
    for i in range(0, n):
        k = int(input())
        arr.append(k)
    print("Largest element in the array : %d " % secondLargestElementInTheArray(arr))

File: Java/Arrays/secondLargestElementInTheArray.java
import java.util.*;

class secondLargestElementInTheArray{

    public static void main(String []args){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of the array : ");
        int n = sc.nextInt();
        ArrayList<Integer> arr = new ArrayList<>();
        System.out.println("Enter the elements of the array : ");
        for(int i=0;i<n;i++){
            int k = sc.nextInt();
            arr.add(k);
        }
        System.out.println("Second largest element : " + secondLargestElementInTheArray(arr));
    }

    public static int secondLargestElementInTheArray(ArrayList<Integer> arr){
        int temp = -1, res = -2;
        for(int i=0;i<arr.size();i++){
            if(temp < arr.get(i)){
                res = temp;
                temp = arr.get(i);
            }
            else if(res < arr.get(i)){
                res = arr.get(i);
            }
        }
        return res;
    }
};